Recruit-Training/Advanced-Recruit-Training/Embedded-Teaser/Training/Core/Src/Training.c

```diff
@@ -1,6 +1,95 @@
 #include "Training.h"
+#include <stdio.h>
 
 void trainingTask(uint8_t* data)
 {
+    uint8_t validity = 0b00000111;
 
+    checkMotors(&validity, data[0], data[1]);
+
+    checkLights(&validity, data[2]);
+
+
+    if (getBit(validity, 0) && getBit(validity, 1)) {
+        outputArray[0] = data[0];
+        outputArray[1] = data[1];
+    }
+    if (getBit(validity, 2)){
+        outputArray[2] = data[2];
+    }
+
+    validData = validity;
+
+}
+
+uint8_t getBit(uint8_t data,int number)
+{
+    return (data & 1<<number);
+}
+
+int getVelocity(uint8_t data)
+{
+    int velocityBits  = getBit(data, 1) + getBit(data, 2) + getBit(data, 3) + getBit(data, 4) + getBit(data, 5);
+    int velocity = velocityBits >> 1;
+
+    if (getBit(data, 6))
+    velocity *=-1;
```

Comment:
This should be indented. You should put this in curly braces even if it's just one line. Do the same for all other lines that this applies to. Other than that, everything else looks good!